File: js/ibh/printNumber.js
export const print = (n) => {
  if (n == 1) {
    console.log(1);
    return;
  }
  print(n - 1);
  console.log(n);
};

export const printRev = (n) => {
  if (n == 1) {
    console.log(1);
    return;
  }
  console.log(n);
  print(n - 1);
};

File: java/src/com/recursion/ibh/PrintNumber.java
package com.recursion.ibh;

public class PrintNumber {
    public void print(int n) {
        if (n == 1){
            System.out.println(1);
            return;
        }
        print(n-1);
        System.out.println(n);
    }

    public void printRev(int n) {
        if (n == 1){
            System.out.println(1);
            return;
        }
        System.out.println(n);
        printRev(n-1);
    }
}
